### PR TITLE
Add auto-push after we commit

### PR DESCRIPTION
**Pull request by taquitos**
- Controlled via param push_after_commit

### CODE REVIEW

File: app/shared/models/git_repo.rb [hunk 1]
```diff
@@ -363,30 +363,32 @@ def reset_hard!(use_global_git_mutex: true)
...
         else
           git.commit(commit_message)
+          push(use_global_git_mutex: false) if push_after_commit
           logger.debug("Done commit_changes! #{self.git_config.full_name} for #{repo_auth.username}")
         end
       end
     end
 
-    def push(repo_auth: self.repo_auth)
-      git_action_with_queue(ensure_block: proc { unset_auth }) do
```
Comment (taquitos) [inline app/shared/models/git_repo.rb]: `unset_auth` happens automatically with `perform_block` method 🎉

File: app/shared/models/git_repo.rb [hunk 2]
```diff
@@ -363,30 +363,32 @@ def reset_hard!(use_global_git_mutex: true)
...
     # TODO: this method isn't actually tested yet
-    def commit_changes!(commit_message: nil, file_to_commit: nil, repo_auth: self.repo_auth)
+    def commit_changes!(commit_message: nil, push_after_commit: true, file_to_commit: nil, repo_auth: self.repo_auth)
       git_action_with_queue do
         logger.debug("Starting commit_changes! #{self.git_config.git_url} for #{repo_auth.username}")
         raise "file_to_commit not yet implemented" if file_to_commit
         commit_message ||= "Automatic commit by fastlane.ci"
 
-        self.setup_author(full_name: repo_auth.full_name, username: repo_auth.username)
+        setup_author(full_name: repo_auth.full_name, username: repo_auth.username)
```
Comment (KrauseFx) [inline app/shared/models/git_repo.rb]: any specific reason for this?

Comment (taquitos) [reply app/shared/models/git_repo.rb]: what part of `this` are you referring to?

Comment (KrauseFx) [reply app/shared/models/git_repo.rb]: Oh sorry, I meant the change of this particular line, being removing the `self` when calling `self.author`

Comment (taquitos) [reply app/shared/models/git_repo.rb]: Ahhh yeah, I'm starting to get used to doing things the `ruby way` which is not using self to refer to stuff.

**Pull request merged**